File: src/front-end/CryptExFrontEnd/src/app/deposit-withdraw/models/bank-account-view-model.ts
export interface BankAccountViewModel {
    id: string;

    iban: string;

    creationDate: string;

    decisionDate: string;

    status: BankAccountStatus;
}

export enum BankAccountStatus {
    notProcessed = -1,
    approved = 1,
    refused = 0
}

File: src/back-end/CryptEx/CryptExApi/Models/ViewModel/BankAccountViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using CryptExApi.Models.Database;

namespace CryptExApi.Models.ViewModel
{
    public class BankAccountViewModel
    {
        public Guid Id { get; set; }

        public string Iban { get; set; }

        public DateTime CreationDate { get; set; }

        /// <summary>When the bank account was approved/refused.</summary>
        public DateTime DecisionDate { get; set; }

        public BankAccountStatus Status { get; set; }

        public static BankAccountViewModel FromBankAccount(BankAccount bankAccount) => new()
        {
            Id = bankAccount.Id,
            Iban = bankAccount.Iban,
            Status = bankAccount.Status,
            CreationDate = bankAccount.CreationDate,
            DecisionDate = bankAccount.DecisionDate
        };
    }
}
